argoverse/utils/make_att_files.py

```diff
@@ -0,0 +1,365 @@
+import glob
+import json
+import os
+import pickle
+from typing import Any, Dict, List, Tuple
+
+import cv2
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.interpolate as interpolate
+
+import argoverse
+import open3d as o3d
+import torch
+import torch.nn.functional as F
+from argoverse.data_loading.argoverse_tracking_loader import ArgoverseTrackingLoader
+from argoverse.utils.json_utils import read_json_file
+
+dict_color: Dict[str, Tuple[float, float, float]] = {}
+dict_color["easy"] = (0.0, 1.0, 0.0)  # green
+dict_color["far"] = (0.0, 0.4, 0.0)  # dark green
+dict_color["occ"] = (0.0, 0.0, 1.0)  # red
+dict_color["fast"] = (0.0, 1.0, 1.0)  # yellow
+dict_color["short"] = (0.8, 0.3, 0.3)  # dark purple
+
+list_attritubes = ["short", "occ", "fast", "far", "easy"]
+
+font = cv2.FONT_HERSHEY_SIMPLEX
+fontScale = 1.2
+fontColor = (255, 255, 255)
+lineType = 3
+visualize = True
+check_track_label_folder = True
+
+
+def save_bev_img(
+    path_output_vis: str,
+    list_bboxes: List[Any],
+    list_difficulty_att: List[Any],
+    dataset_name: str,
+    log_id: str,
+    lidar_timestamp: int,
+    pc: np.ndarray,
+    egovehicle_to_city_se3: Any,
+) -> None:
+    """
+    Plot results on bev images and save 
+    """
+    image_size = 2000
+    image_scale = 10
+    img = np.zeros((image_size, image_size, 3))
+    pc = pc * image_scale
+    pc[:, 0] += int(image_size / 2)
+    pc[:, 1] += int(image_size / 2)
+    pc = pc.astype("int")
+
+    ind_valid = np.logical_and(
```

Comment:
you can actually do this with `np.logical_and.reduce()` and just pass in a list of predicates, instead of chaining several `np.logical_and()` calls together.

>>> `import numpy as np`
>>> `x = np.array([1,2,3,4,5])`
>>> `np.logical_and.reduce( [x>=2, x <=4] )`
array([False,  True,  True,  True, False], dtype=bool)
>>>